src/Xamarin.Android.Build.Tasks/Xamarin.Android.Tooling.targets

```diff
@@ -43,20 +43,22 @@ projects.
   </Target>
 
   <PropertyGroup Condition=" '$(UsingAndroidNETSdk)' == 'True' ">
-    <_ResolveSdksDependsOnTargets>ResolveFrameworkReferences</_ResolveSdksDependsOnTargets>
+    <_ResolveSdksDependsOnTargets>ResolveTargetingPackAssets</_ResolveSdksDependsOnTargets>
   </PropertyGroup>
   <PropertyGroup Condition=" '$(UsingAndroidNETSdk)' != 'True' ">
     <_ResolveSdksDependsOnTargets>_GetReferenceAssemblyPaths</_ResolveSdksDependsOnTargets>
   </PropertyGroup>
 
   <Target Name="_ResolveSdks" DependsOnTargets="$(_ResolveSdksDependsOnTargets)">
+    <!-- When using .NET 5, provide the path to Mono.Android.dll e.g.
+            `packages\microsoft.android.ref\10.0.100-ci.master.22\ref\net5.0\`
+    -->
+    <ItemGroup Condition=" '$(UsingAndroidNETSdk)' == 'True' ">
+      <_XATargetFrameworkDirectory Include="@(Reference->'%(RootDir)%(Directory)')" Condition=" '%(FileName)%(Extension)' == 'Mono.Android.dll' " />
+    </ItemGroup>
     <PropertyGroup>
       <_AndroidAllowMissingSdkTooling Condition=" '$(_AndroidAllowMissingSdkTooling)' == '' ">False</_AndroidAllowMissingSdkTooling>
-      <!-- When using .NET 5, provide a list of paths to all framework reference directories, e.g.
-            `packages\xamarin.android.app.ref\10.3.99.160\ref\netcoreapp5.0\` and
-            `C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Ref\5.0.0-preview.2.20160.6\ref\netcoreapp5.0\`
-      -->
-      <_XATargetFrameworkDirectories Condition=" '$(UsingAndroidNETSdk)' == 'True' ">@(ResolvedFrameworkReference->'%(TargetingPackPath)\ref\$(TargetFramework)')</_XATargetFrameworkDirectories>
+      <_XATargetFrameworkDirectories Condition=" '$(UsingAndroidNETSdk)' == 'True' ">@(_XATargetFrameworkDirectory)</_XATargetFrameworkDirectories>
```

Comment:
Does it work to do this all in one line?
```xml
<_XATargetFrameworkDirectories Condition=" '$(UsingAndroidNETSdk)' == 'True' And '%(FileName)%(Extension)' == 'Mono.Android.dll'  ">@(Reference->'%(RootDir)%(Directory)')</_XATargetFrameworkDirectories>
```
Just so we don't create a `@(_XATargetFrameworkDirectory)` item group that isn't used anywhere else.

Reply:
I've had to rework this anyway, as `$(_XATargetFrameworkDirectory)` needs to contain paths to both Mono.Android.dll and mscorlib.dll.

Reply:
I think I need to use a new ItemGroup here to create an intersection of `@(ResolvedTargetingPack)` and `@(FrameworkReference)`. If I use this same condition in a property only the first ItemGroup entry that meets the condition is used as the property value, rather than the full set of matches.